osaca/semantics/marker_utils.py

```diff
@@ -147,29 +150,30 @@ def find_marked_section(
                 destination = line.operands[1 if not reverse else 0]
                 # instruction pair matches, check for operands
                 if (
-                    "immediate" in source
-                    and parser.normalize_imd(source.immediate) == mov_vals[0]
-                    and "register" in destination
-                    and parser.get_full_reg_name(destination.register) == mov_reg
+                    isinstance(source, ImmediateOperand)
+                    and parser.normalize_imd(source) == mov_vals[0]
+                    and isinstance(destination, RegisterOperand)
+                    and parser.get_full_reg_name(destination) == mov_reg
                 ):
                     # operands of first instruction match start, check for second one
                     match, line_count = match_bytes(lines, i + 1, nop_bytes)
                     if match:
                         # return first line after the marker
                         index_start = i + 1 + line_count
                 elif (
-                    "immediate" in source
-                    and parser.normalize_imd(source.immediate) == mov_vals[1]
-                    and "register" in destination
-                    and parser.get_full_reg_name(destination.register) == mov_reg
+                    isinstance(source, ImmediateOperand)
+                    and parser.normalize_imd(source) == mov_vals[1]
+                    and isinstance(destination, RegisterOperand)
+                    and parser.get_full_reg_name(destination) == mov_reg
                 ):
                     # operand of first instruction match end, check for second one
                     match, line_count = match_bytes(lines, i + 1, nop_bytes)
                     if match:
                         # return line of the marker
                         index_end = i
         except TypeError:
-            print(i, line)
+            pass
```

Comment:
I think keeping the printing of the line that failed is helpful and should be kept.

Reply:
https://github.com/RRZE-HPC/OSACA/commit/1c0708e750c5f08bf5edc5f9b18a4cacc2437fe0#diff-30356bac9a86f5ca82fdce3358b4819ceefb05039c93657094dd54942da50315